conans/client/installer.py

```diff
@@ -426,6 +433,7 @@ def _build(self, nodes_by_level, keep_build, root_node, graph_info, remotes, bui
                         self._binaries_analyzer.reevaluate_node(node, remotes, build_mode, update)
                         if node.binary == BINARY_MISSING:
                             self._raise_missing([node])
+                        # TODO: Check if BINARY_INVALID?
```

Comment:
I'd say this is not needed. This will happen only for nodes that match `node.binary == BINARY_UNKNOWN` and this only happens when `package_id` is `PACKAGE_ID_UNKNOWN`... which is never set when the binary is invalid. I'd say that a reevaluation/repropagation will never turn an invalid into an unknown.